Package.swift

```diff
@@ -20,10 +20,12 @@ let package = Package(
     products: [
         .library(name: "SpeziOpenAI", targets: ["SpeziOpenAI"]),
         .library(name: "SpeziSpeechRecognizer", targets: ["SpeziSpeechRecognizer"]),
-        .library(name: "SpeziSpeechSynthesizer", targets: ["SpeziSpeechSynthesizer"])
+        .library(name: "SpeziSpeechSynthesizer", targets: ["SpeziSpeechSynthesizer"]),
+        .library(name: "SpeziLocalLLM", targets: ["SpeziLocalLLM"])
     ],
     dependencies: [
         .package(url: "https://github.com/MacPaw/OpenAI", .upToNextMinor(from: "0.2.4")),
+        .package(url: "https://github.com/ggerganov/llama.cpp", branch: "b1422"),
```

Comment:
We will probably need to use a concrete version or a commit hash here as branch definitions are not allowed when we want to use the package in any other swift package: https://developer.apple.com/documentation/packagedescription/package/dependency/requirement-swift.enum/branch(_:)#discussion

Reply:
To be honest, I'm not sure how to resolve that issue. 
An exact commit hash could be specified via the `.package(url:revision:)` (see https://developer.apple.com/documentation/packagedescription/package/dependency/package(url:revision:)) but this option has the same limitation as described by you with the `.branch` modifier (see https://developer.apple.com/documentation/packagedescription/package/dependency/requirement-swift.enum/revision(_:)). 
Of course, we could fork `llama.cpp` and add semantic versioning but I don't think that is a good solution long term. The other option would be that `llama.cpp` adopts semantic versioning, but apperently the interest isn't really there and they follow other versioning standards: https://github.com/ggerganov/llama.cpp/discussions/2737 and https://github.com/ggerganov/llama.cpp/pull/2732

Reply:
@philippzagar Thanks for the additional context!
You are right, `revision` will also not work for us.

The very short-term version would be to fork llama.cpp, but in the long run, it would be ideal to reach out, share the motivation behind it (Swift Package Manager, which they actually support), and then maybe create a small PR to llama.cpp that introduces a solution:
I think the best middle ground would be to propose a change of versioning scheme to `0.BUILD_NUMBER.0`. with `BUILD_NUMBER` being the current sequential versioning that they use. This would at least allow one to use the package with semantic versioning, and it would communicate that each change might be a breaking change, as typically done with 0 versions.

If there is no intention to adopt this, we could use GitHub Actions always to update our fork and just map any new releases to ones using semantic versioning and let the repo update itself very frequently but in an automated fashion.

Reply:
Here is a fork that also provides you access to tag and/or automate the setup: https://github.com/StanfordBDHG/llama.cpp

Reply:
Perfect, thank you for creating the fork, that seems like a good solution short term 🚀
I will reach out to the llama.cpp guys in the coming days and will implement the respective changes so that we can get rid of our StanfordBDHG fork 👍

Reply:
Here the PR in the `llama.cpp` repo: https://github.com/ggerganov/llama.cpp/pull/3904

Reply:
Sadly, I just noticed that semantically versioned SPM dependencies don't allow `unsafeFlags(_:)` (see see in the [Swift community](https://forums.swift.org/t/question-about-contains-unsafe-build-flags-in-spm/47677) and on [StackOverflow](https://stackoverflow.com/questions/67889931/the-package-product-mylibrary-cannot-be-used-as-a-dependency-of-this-target-be)), which are used in `llama.cpp` (see https://github.com/ggerganov/llama.cpp/blob/master/Package.swift).
I just posted a comment in the `llama.cpp` PR if there is any chance that those flags could be removed / replaced.

Reply:
Thank you for the PR and the additional comment there.

Let's see what they respond with. 

Maybe it might be a requirement to add the build flag to the consuming project and not in the package itself?

Reply:
Adding the build flag to the consuming project isn't a feasible option I guess as the standalone package probably wouldn't even compile anymore. Therefore, I don't think this is a good option (and may not even work)
=> I hope there is some way that `llama.cpp` can get rid of the `unsafeFlags(_:)`, otherwise it may be quite a tricky situation...

Reply:
Thank you for the additional context; makes sense. This seems to be a blocker ...

Does llama.cpp currently stop compiling if we remove the flag in our fork?

Reply:
Resolved via binary SPM package within https://github.com/StanfordBDHG/llama.cpp